### ISSUE TITLE
Multiple Group/ ANOVA-like Comparisons in Differential Labelling Experiments? 

### ISSUE THREAD

**Issue by TJGross7285** [jcapelladesto/geoRge#15]
Hi Jordi,

Thank you for this fantastic software tool. I had a question/ comment about the statistical tests used for differential labeling analysis. T-tests make perfect sense for comparing labelled to unlabelled specimens. For differential labelling experiments, however, it would be fantastic to support ANOVA-like comparisons across more than two experimental groups. Is this for any reason unfeasible with the current implementation of geoRge? Am I missing an existing framework to carry out this type of analysis? I would worry about false discoveries emerging from carrying out all pairwise comparisons across all experimental groups. Thanks!

-Tom

**Comment by jcapelladesto** [comment]
Hi Tom,

Any type of different of analysis should be feasible as long as it is possible to code. ANOVA analysis though, require setting up the factors, and interactions (if two- or three-way ANOVA), a one-way ANOVA is not different from a T-test. 
At the moment there is no implementation of ANOVA ready for geoRge. But, there is two-types of statistical analysis performed in geoRge:

- In PuInc_seeker(), which, for each xcms-feature, performs a Welch's T-test  between Unlabelled and Labelled samples (of the same biological group). For example: C12_control vs C13_control and C12_case vs C13_case, separately.

- In label_compare(), which, for each reported isotopologue, performs a Welch's T-test between different sample groups. Following the example above: % labelling_control vs % labelling_case.

Could you explain, which function were you thinking of applying the ANOVA?

Thank you for your suggestion,

Jordi

**Comment by TJGross7285** [comment]
Thanks for your response Jordi

PuInc_seeker() seems well-suited to two-group comparisons through T-tests (comparing labelled to unlabelled by biological group). I don't see a need to implement ANOVA there. In contrast, label_compare() might benefit greatly from one-way comparisons across more than % labelling_control vs % labelling_case. It is my understanding that T-tests would only support two group comparisons here (in contrast to one-way ANOVA which can compare across sample group levels > 2).  

For example, the Alzheimer's (AD) field has identified more and more clinical stages prior to frank dementia but more advanced than healthy controls. It would be very interesting to identify blood metabolite features which are significantly differentially labelled across some or all of these phenotypes. Without exhausting all pairwise group comparisons, a one-way ANOVA approach could provide an initial, omnibus comparison feature-by-feature (e.g.,  % labelling_control vs % labelling_preclinicalAD vs %labelling_prodromalAD vs %labelling_clinicalAD). 

Is this example helpful to explain my thinking and experimental rationale?    

Best,
Tom

**Comment by jcapelladesto** [comment]
I completely agree with your point. I will look into it and update the package, a one-way ANOVA should be easily implementable.
Are you particularly interested in this?

Jordi

**Comment by TJGross7285** [comment]
That sounds fantastic Jordi!

I certainly am interested in this specific functionality. I anticipate it being a key strength in some human-participants studies I am beginning to plan for my post-doc. I look forward to prominently featuring and citing geoRge in that proposed work. 

If you are interested, I can update you regarding where these experiments go. Stable-isotope-labeled untargeted metabolomics, to me, appears to have tremendous unrealized biomedical potential in the discovery phase of human clinical research. I'm very interested to consider how the evolution of software tools like geoRge might support effective experimental designs in these adult clinical populations.  

Tom 

Edit: It looks like `stats::oneway.test(var.equal = FALSE)` generalizes Welch's correction for greater than two-group comparisons. This complements the existing _t_-test implementations well.
